test/development/basic/hmr.test.ts

```diff
@@ -362,6 +362,38 @@ describe.each([[''], ['/docs']])(
             }
           }
         })
+
+        it('should not full reload when nonlatin characters are used', async () => {
+          let browser = null
+          const pagePath = join('pages', 'hmr', 'nonlatin.js')
+          const originalContent = await next.readFile(pagePath)
+          try {
+            browser = await webdriver(next.url, basePath + '/hmr/nonlatin')
+            const timeOrigin = await browser.eval('performance.timeOrigin')
+            const editedContent = originalContent.replace(
+              '<div>テスト</div>',
+              '<h1>テスト</h1>'
+            )
+
+            // Change the page
+            await next.patchFile(pagePath, editedContent)
+
+            // wait for 5 seconds
+            await waitFor(5000)
```

Comment:
Instead of a timeout here (timeouts are a huge source of the current nextjs test flakiness), can we instead wait for the content of the page to match the patched version? That avoids a hardcoded timeout, and also verifies that HMR actually did something.